Source/SwiftLintBuiltInRules/Rules/Idiomatic/UnusedEnumeratedRule.swift

```diff
@@ -44,17 +56,87 @@ private extension UnusedEnumeratedRule {
                 return
             }
 
-            let position: AbsolutePosition
-            let reason: String
-            if firstTokenIsUnderscore {
-                position = firstElement.positionAfterSkippingLeadingTrivia
-                reason = "When the index is not used, `.enumerated()` can be removed"
+            addViolation(
+                zeroPosition: firstTokenIsUnderscore ? firstElement.positionAfterSkippingLeadingTrivia : nil,
+                onePosition: firstTokenIsUnderscore ? nil : secondElement.positionAfterSkippingLeadingTrivia
+            )
+        }
+
+        override func visit(_ node: FunctionCallExprSyntax) -> SyntaxVisitorContinueKind {
+            guard node.isEnumerated,
+                  let trailingClosure = node.parent?.parent?.as(FunctionCallExprSyntax.self)?.trailingClosure,
+                  node.parent?.as(MemberAccessExprSyntax.self)?.declName.baseName.text != "filter"
+            else {
+                return .visitChildren
+            }
+
+            if let parameterClause = trailingClosure.signature?.parameterClause {
+                guard let parameterClause = parameterClause.as(ClosureShorthandParameterListSyntax.self),
+                      parameterClause.count == 2,
+                      let firstElement = parameterClause.firstParameter,
+                      let secondElement = parameterClause.lastParameter,
+                      case let firstTokenIsUnderscore = firstElement.isUnderscore,
+                      case let lastTokenIsUnderscore = secondElement.isUnderscore,
+                      firstTokenIsUnderscore || lastTokenIsUnderscore
+                else {
+                    return .visitChildren
+                }
+
+                addViolation(
+                    zeroPosition: firstTokenIsUnderscore ? firstElement.positionAfterSkippingLeadingTrivia : nil,
+                    onePosition: firstTokenIsUnderscore ? nil : secondElement.positionAfterSkippingLeadingTrivia
+                )
             } else {
-                position = secondElement.positionAfterSkippingLeadingTrivia
-                reason = "When the item is not used, `.indices` should be used instead of `.enumerated()`"
+                self.trailingClosure = trailingClosure
+            }
+
+            return .visitChildren
+        }
+
+        override func visitPost(_ node: ClosureExprSyntax) {
```

Comment:
This not going to work for closures in closures. I recommend to go with a separate visitor only collecting appearances of `$0`, `$0.element`, ...

Please also add tests.

Reply:
Cool - will take a bit of time to work this out

Reply:
Do you have an example that you think will fail @SimplyDanny ?

Reply:
Without trying:

```swift
list.enumerated().map {
    $1.enumerated().forEach { print($0, $1) }
    return $0
}
```

I assume this would trigger the rule on the outer `enumerated()`, even though it's actually validly used.

Reply:
ok - so I think I have it working for closures within closures, indefinitely. The basic idea is to keep a stack that tracks   the zero and one position for each closure, and then push and pop the data structures onto the stack as we encounter and depart each closure.

It feels to me like it adds a fair bit of complexity for something that is a bit of an edge case, but it does pass all the tests and some new ones. First pass of code is on a PR targeting this branch at 

https://github.com/mildm8nnered/SwiftLint/pull/11

If you think that generally looks good-ish, I'll tidy it up and merge it into this branch.

Reply:
I added some comments in the other PR.

Reply:
Thanks! I've either adopted them wholesale, or replied inline on this PR ...